### PR TITLE
Convert UtcDate model to Kotlin

### PR DESCRIPTION
**Pull request by anticafe**

### CODE REVIEW

File: app/src/main/java/org/wikipedia/feed/model/UtcDate.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    private val cal: Calendar = Calendar.getInstance(TimeZone.getTimeZone("UTC"))
+    private val year: String
+    private val month: String
+    private val date: String
+
+    fun baseCalendar(): Calendar {
+        return cal
+    }
+
+    fun year(): String {
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/feed/model/UtcDate.kt:16]: None of these getter methods are necessary. That's one of the benefits of converting to Kotlin.

File: app/src/main/java/org/wikipedia/feed/model/UtcDate.kt [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    init {
+        cal.add(Calendar.DATE, -age)
+        year = cal[Calendar.YEAR].toString()
+        month = pad((cal[Calendar.MONTH] + 1).toString())
+        date = pad(cal[Calendar.DATE].toString())
+    }
+
+    companion object {
+        private fun pad(value: String): String {
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/feed/model/UtcDate.kt:36]: This function is no longer necessary. Kotlin has this function built in.

File: app/src/main/java/org/wikipedia/feed/model/UtcDate.kt [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        return month
+    }
+
+    fun date(): String {
+        return date
+    }
+
+    init {
+        cal.add(Calendar.DATE, -age)
+        year = cal[Calendar.YEAR].toString()
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/feed/model/UtcDate.kt:30]: These initialization bits don't need to be here. They can be made into `get()` properties inline with the actual fields.

**Review by dbrant — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/wikipedia/feed/model/UtcDate.kt (hunk 3 above)
Comment (anticafe) [reply app/src/main/java/org/wikipedia/feed/model/UtcDate.kt]: Updated via 4ae3b1e

File: app/src/main/java/org/wikipedia/feed/model/UtcDate.kt (hunk 2 above)
Comment (anticafe) [reply app/src/main/java/org/wikipedia/feed/model/UtcDate.kt]: Thanks for suggestion. IDK Kotlin has built-in `padStart` method. 
Updated via 4ae3b1e

File: app/src/main/java/org/wikipedia/feed/model/UtcDate.kt (hunk 1 above)
Comment (anticafe) [reply app/src/main/java/org/wikipedia/feed/model/UtcDate.kt]: Updated via 4ae3b1e

**Pull request merged**